src/Umbraco.Web.UI.Client/src/common/directives/components/umbmediagrid.directive.js

```diff
@@ -120,7 +120,7 @@ Use this directive to generate a thumbnail grid of media items.
                     setOriginalSize(item, itemMaxHeight);
 
                     // remove non images when onlyImages is set to true
-                    if (scope.onlyImages === "true" && item.isFolder === true) {
+                    if (scope.onlyImages === "true" && item.isFolder === false && item.thumbnail == null) {
```

Comment:
Don't we still also want to check `isFolder === true`? If that is the case then it should definitely be ignored right?

maybe like?

```js
if (scope.onlyImages === "true" && (item.isFolder === true || (item.isFolder === false && item.thumbnail == null)))
```

Reply:
Agreed!

Reply:
If we include item.isFolder === true, all folders will be removed, thereby preventing navigation down the folder tree?